tools/ci_build/github/azure-pipelines/win-gpu-ci-pipeline.yml

```diff
@@ -13,7 +13,8 @@ stages:
         BuildConfig: 'RelWithDebInfo'
         EnvSetupScript: setup_env_cuda_11.bat
         buildArch: x64
-        additionalBuildFlags: --enable_pybind --build_java --build_nodejs --use_cuda --cuda_version=11.4 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4"  --gen_doc validate --enable_cuda_profiling --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=52
+        additionalBuildFlags: --enable_pybind --build_java --build_nodejs --use_cuda --cuda_version=11.4 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4"  --gen_doc validate --enable_cuda_profiling --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=52 --use_dml
```

Comment:
Previously, I found DML and CUDA couldn't coexist. I remember memory copy wasn't implemented for buffers between them.

Reply:
Hmm, yeah, the intention isn't to ever actually *execute* both CUDA and DML from this build, just to utilize the operator registration code to gather the kernel operator list. Can you think of a cleaner approach?

Reply:
Create a new, separated build job for serving this purpose?

Reply:
I'm okay with that if you are :). (though, I might need some hand holding to know what things to update)

Reply:
That's pretty expensive isn't it? It needs to have the CPU, CUDA and DML EPs which is a lot to build to generate some doco. 

If the tests don't enable DML is there an issue including DML in the build?

Reply:
Are the bits produced by the Windows GPU pipeline stage the official ones released on Nuget? If so, my current change would unintentionally impact customer bits 🤔. Currently we have https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime.Gpu (or really *OnnxRuntime.CUDA*) and https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime.DirectML, and I wonder if this would put both DML and CUDA into the same ORT, giving users the impression that both EP's can work together (I mean maybe they can work together, with some work and going through CPU RAM as a conduit, but that's getting far away from the intention of this PR).

Reply:
No. They are not.

Reply:
That said, maybe it's better to have a separate stage that runs in parallel that purely does the build to generate doco so it's a cleaner setup than hacking `--gen_doc` in here.

Reply:
Added new documentation stage. 🤞